traveler-ios-ui/TravelerKitUI/ViewControllers/BookingItemSearch/BookingItemSearchSortFilterViewController.swift

```diff
@@ -0,0 +1,191 @@
+        delegate?.bookingItemSearchSortFilterController(self, didSubmit: searchQuery!)
+    }
+    
+    func resetData() {
+        currentSort = defaultSort
+        
+        if let minFacetsPrice = currentSearchFacets?.minPrice.valueInBaseCurrency, let maxFacetsPrice = currentSearchFacets?.maxPrice.valueInBaseCurrency, minFacetsPrice <= maxFacetsPrice {
+            currentRange = minFacetsPrice...maxFacetsPrice
+        } else {
+            currentRange = fullRange
+        }
+    }
+    
+    @IBAction func reset() {
+        resetData()
+        tableView.reloadData()
+    }
+    
+    public override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    
+    public override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return 1
+        case 1:
+            return 1
+        default:
+            Log("Unhandled section; this should not happen!")
+            return 0
+        }
+    }
+    
+    public override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
+        case 0:
+            return "Sort"
+        case 1:
+            return "Filters"
+        default:
+            Log("Unhandled section; this should not happen!")
+            return ""
+        }
+    }
+    
+    public override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: CellIdentifiers.bookingItemSearchSortFilter, for: indexPath) as! BookingItemSearchSortFilterCell
+        cell.selectionStyle = .none
+        
+        switch indexPath {
+        case Paths.sort:
+            cell.titleLabel.text = "Order by"
+            cell.optionLabel.text = currentSort!.data.title
+        case Paths.price:
+            cell.titleLabel.text = "Price"
+            cell.optionLabel.text = String(format: "$%.2f - $%.2f", currentRange!.lowerBound, currentRange!.upperBound)
+        default:
+            Log("Unhandled sort/filter option.")
+        }
+        
+        return cell
+    }
+    
+    public override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        switch indexPath {
+        case Paths.sort:
+            performSegue(withIdentifier: "sortSegue", sender: self)
+        case Paths.price:
+            performSegue(withIdentifier: "filterPriceSegue", sender: self)
+        default:
+            Log("Unhandled sort/filter option.")
+        }
+    }
+}
+
+extension BookingItemSearchSortFilterViewController: BookingItemSearchSortViewControllerDelegate {
+    public func bookingItemSearchSortViewController(_ controller: BookingItemSearchSortViewController, didUpdate sortOption: ProductItemSort) {
+        controller.dismiss(animated: true, completion: { [unowned self] in
```

Comment:
My opinion is that this controller should dismiss itself. If the "apply" button has been clicked it is safe to assume that, all the work meant to be done by that controller is done. Therefore it should call the delegate and then dismiss itself.

Reply:
This is done more so to ensure that the previous controller is dismissed first prior to attempting to reload the tableView rows (UI Operation).  You always want to make sure the current view is actually in view prior to doing UI operations.